client/image_tag.go

```diff
@@ -12,23 +12,40 @@ import (
 )
 
 // ImageTag tags an image in the docker host
-func (cli *Client) ImageTag(ctx context.Context, imageID, ref string) error {
-	distributionRef, err := distreference.ParseNamed(ref)
+func (cli *Client) ImageTag(ctx context.Context, source, target string) error {
+	// For Usage:	docker image tag SOURCE_IMAGE[:TAG] TARGET_IMAGE[:TAG]
+	// If there is a reason for checking ARGET_IMAGE[:TAG] ,
+	// it's also suit for SOURCE_IMAGE[:TAG], not only one of them, all both.
+	_, err := checkDisReference(source)
 	if err != nil {
-		return fmt.Errorf("Error parsing reference: %q is not a valid repository/tag", ref)
+		return err
 	}
 
-	if _, isCanonical := distributionRef.(distreference.Canonical); isCanonical {
-		return errors.New("refusing to create a tag with a digest reference")
+	targetDisRef, err := checkDisReference(target)
+	if err != nil {
+		return err
 	}
 
-	tag := reference.GetTagFromNamedRef(distributionRef)
+	tag := reference.GetTagFromNamedRef(targetDisRef)
 
 	query := url.Values{}
-	query.Set("repo", distributionRef.Name())
+	query.Set("repo", targetDisRef.Name())
 	query.Set("tag", tag)
 
-	resp, err := cli.post(ctx, "/images/"+imageID+"/tag", query, nil, nil)
+	resp, err := cli.post(ctx, "/images/"+source+"/tag", query, nil, nil)
 	ensureReaderClosed(resp)
 	return err
 }
+
+func checkDisReference(ref string) (distreference.Named, error) {
+	distributionRef, err := distreference.ParseNamed(ref)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing reference: %q is not a valid repository/tag", ref)
+	}
+
+	if _, isCanonical := distributionRef.(distreference.Canonical); isCanonical {
+		return nil, errors.New("refusing to create a tag with a digest reference")
```

Comment:
Because two different checks now have to be done, I don't think this wrapper function `checkDisReference()` is needed; I suggest to just do these checks inline